### PR TITLE
needsReduceOnly logic and tests

### PR DESCRIPTION
**Pull request by johnqh**

### CODE REVIEW

File: src/commonMain/kotlin/exchange.dydx.abacus/output/input/TradeInput.kt [hunk 1]
```diff
@@ -70,6 +69,7 @@ data class TradeInputOptions(
     val timeInForceOptions: IList<SelectionOption>?,
     val goodTilUnitOptions: IList<SelectionOption>,
     val executionOptions: IList<SelectionOption>?,
+    val reduceOnlyPromptStringKey: String?
```
Comment (mike-dydx) [inline src/commonMain/kotlin/exchange.dydx.abacus/output/input/TradeInput.kt:72]: @johnqh how come we have this for `reduceOnly` but not `postOnly`?

File: src/commonMain/kotlin/exchange.dydx.abacus/calculator/TradeInputCalculator.kt [hunk 2]
```diff
@@ -1027,6 +1039,11 @@ internal class TradeInputCalculator(
             } else {
                 options.safeSet("maxLeverage", null)
             }
+            if (parser.asBool(options["needsReduceOnly"]) == true) {
+                options.safeSet("reduceOnlyPromptStringKey", null)
+            } else {
+                options.safeSet("reduceOnlyPromptStringKey", reduceOnlyPromptFromTrade(trade))
+            }
```
Comment (mike-dydx) [inline src/commonMain/kotlin/exchange.dydx.abacus/calculator/TradeInputCalculator.kt:1046]: is this reversed? if `needsReduceOnly` is true, I'm thinking  `reduceOnlyPromptStringKey` should be non-nil so client knows what to display

File: src/commonMain/kotlin/exchange.dydx.abacus/calculator/TradeInputCalculator.kt [hunk 3]
```diff
@@ -827,32 +836,32 @@ internal class TradeInputCalculator(
...
-                trade,
-                "execution"
-            )
-        )
-        return when (execution) {
-            "FOK", "IOC" -> reduceOnlyField()
-            else -> null
-        }
-    }
+//
```
Comment (mike-dydx) [inline src/commonMain/kotlin/exchange.dydx.abacus/calculator/TradeInputCalculator.kt:839]: delete or intentionally commented?

**Review by jaredvu — APPROVED** [review]
Which ver of v4-localization do we need with this ver of v4-abacus?

File: src/commonMain/kotlin/exchange.dydx.abacus/calculator/TradeInputCalculator.kt (hunk 2 above)
Comment (mike-dydx) [reply src/commonMain/kotlin/exchange.dydx.abacus/calculator/TradeInputCalculator.kt:1020]: ah i see. Small suggestion then to rename to something like `reduceOnlyUnavailablePromptStringKey` but not strong feelings on that

**Review by mike-dydx — APPROVED** [review]

**Pull request merged**